File: lemminx-dayz-ce/src/test/java/io/github/rvost/lemminx/dayz/LimitsDefinitionUserModelTests.java
package io.github.rvost.lemminx.dayz;

import com.google.common.collect.ImmutableBiMap;
import io.github.rvost.lemminx.dayz.model.LimitsDefinitionUserModel;
import io.github.rvost.lemminx.dayz.utils.DocumentUtils;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.nio.file.Path;
import java.util.Map;

public class LimitsDefinitionUserModelTests {

    @Test
    public void testEmptyCfgLimitsDefinitionsUser() throws Exception {
        var url = getClass().getClassLoader().getResource("cfglimitsdefinitionuser/empty.xml");
        var path = Path.of(url.toURI());

        var flags = DocumentUtils.tryParseDocument(path)
                .map(LimitsDefinitionUserModel::getUserLimitsDefinitions)
                .orElse(Map.of());

        Assertions.assertEquals(0, flags.get("usage").size());
        Assertions.assertEquals(0, flags.get("value").size());
    }

    @Test
    public void testSimpleCfgLimitsDefinitionsUser() throws Exception {
        var url = getClass().getClassLoader().getResource("cfglimitsdefinitionuser/simple.xml");
        var path = Path.of(url.toURI());

        var flags = DocumentUtils.tryParseDocument(path)
                .map(LimitsDefinitionUserModel::getUserLimitsDefinitions)
                .orElse(Map.of());

        Assertions.assertEquals(2, flags.size());
        Assertions.assertTrue(flags.containsKey("usage"));
        Assertions.assertTrue(flags.get("usage").contains("TownVillage"));
        Assertions.assertTrue(flags.containsKey("value"));
        Assertions.assertTrue(flags.get("value").contains("Tier12"));
    }

    @Test
    public void testSimpleUserFlagDefinitions() throws Exception {
        var url = getClass().getClassLoader().getResource("cfglimitsdefinitionuser/simple.xml");
        var path = Path.of(url.toURI());

        var flags = DocumentUtils.tryParseDocument(path)
                .map(LimitsDefinitionUserModel::getUserFlags)
                .orElse(ImmutableBiMap.of());

        Assertions.assertEquals(2, flags.size());
        Assertions.assertTrue(flags.containsKey("TownVillage"));
        Assertions.assertEquals(2, flags.get("TownVillage").size());
    }
}

File: lemminx-dayz-ce/src/test/kotlin/io/github/rvost/lemminx/dayz/model/LimitsDefinitionUserModelTests.kt
package io.github.rvost.lemminx.dayz.model

import io.github.rvost.lemminx.dayz.utils.DocumentUtils
import org.eclipse.lemminx.dom.DOMDocument
import org.junit.jupiter.api.Test
import java.nio.file.Path
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class LimitsDefinitionUserModelTests {
    @Test
    fun testEmptyCfgLimitsDefinitionsUser() {
        val url = javaClass.classLoader.getResource("cfglimitsdefinitionuser/empty.xml")
        val path = Path.of(url.toURI())

        val flags = DocumentUtils.tryParseDocument(path)
            .map { doc: DOMDocument? -> LimitsDefinitionUserModel.getUserLimitsDefinitions(doc) }
            .orElseThrow()

        assertEquals(0, flags["usage"]!!.size)
        assertEquals(0, flags["value"]!!.size)
    }

    @Test
    fun testSimpleCfgLimitsDefinitionsUser() {
        val url = javaClass.classLoader.getResource("cfglimitsdefinitionuser/simple.xml")
        val path = Path.of(url.toURI())

        val flags = DocumentUtils.tryParseDocument(path)
            .map { doc: DOMDocument? -> LimitsDefinitionUserModel.getUserLimitsDefinitions(doc) }
            .orElseThrow()

        assertEquals(2, flags.size)
        assertTrue(flags["usage"]!!.contains("TownVillage"))
        assertTrue(flags["value"]!!.contains("Tier12"))
    }

    @Test
    fun testSimpleUserFlagDefinitions() {
        val url = javaClass.classLoader.getResource("cfglimitsdefinitionuser/simple.xml")
        val path = Path.of(url.toURI())

        val flags = DocumentUtils.tryParseDocument(path)
            .map { doc: DOMDocument? -> LimitsDefinitionUserModel.getUserFlags(doc) }
            .orElseThrow()

        assertEquals(2, flags.size)
        assertTrue(flags.contains("TownVillage"))
        assertEquals(2, flags["TownVillage"]!!.size)
    }
}
